grpc-api/src/main/java/io/deephaven/grpc_api/arrow/FlightServiceGrpcImpl.java

```diff
@@ -394,10 +395,17 @@ private void process(final MessageInfo mi) {
             final int numColumns = resultTable.getColumnSources().size();
             final BarrageMessage msg = new BarrageMessage();
             final RecordBatch batch = (RecordBatch) mi.header.header(new RecordBatch());
+
+            final MutableInt bufferOffset = new MutableInt();
             final Iterator<ChunkInputStreamGenerator.FieldNodeInfo> fieldNodeIter =
                     new FlatBufferIteratorAdapter<>(batch.nodesLength(), i -> new ChunkInputStreamGenerator.FieldNodeInfo(batch.nodes(i)));
             final Iterator<ChunkInputStreamGenerator.BufferInfo> bufferInfoIter =
-                    new FlatBufferIteratorAdapter<>(batch.buffersLength(), i -> new ChunkInputStreamGenerator.BufferInfo(batch.buffers(i)));
+                    new FlatBufferIteratorAdapter<>(batch.buffersLength(), i -> {
+                        int offset = LongSizedDataStructure.intSize("BufferInfo", batch.buffers(i).offset());
+                        offset -= bufferOffset.getAndAdd(offset);
+                        final int length = LongSizedDataStructure.intSize("BufferInfo", batch.buffers(i).length());
+                        return new ChunkInputStreamGenerator.BufferInfo(offset, length);
+                    });
```

Comment:
Because I'm awesome and duplicated DoGet and subscription paths; this looks wrong based on your +length fix in the other location.

Reply:
Will fix, and am working on tests that will confirm this as part of the build.

Reply:
Seconded, this clearly does not match the version in `BarrageStreamReader`. That said, we may be over-complicating matters if our adjusted offset is always `0`....